src/summarycode/summarizer.py

```diff
@@ -502,11 +503,29 @@ def package_summarizer(resource, children, keep_details=False):
         logger_debug('package_summarizer: for:', resource,
                      'packages are:', packs)
 
+    package_purls = []
+    for package in packages:        
+        if package.get("purl", False):
+            package_purls.append(package["purl"])
+
     # Collect direct children packages summary
     for child in children:
         child_summaries = get_resource_summary(child, key='packages', as_attribute=False) or []
-        packages.extend(child_summaries)
-
+        for child_summary in child_summaries:
+            values = [child_summary['value']] * child_summary['count']
+            package_purls.extend(values)
+ 
     # summarize proper
-    set_resource_summary(resource, key='packages', value=packages, as_attribute=False)
-    return packages
+    packages_counter = summarize_licenses(package_purls)
+    summarized = sorted_counter(packages_counter)
+    set_resource_summary(resource, key="packages", value=summarized, as_attribute=keep_details)
+
+    return summarized
+    
+
+def summarize_packages(package_purls):
+    """
+    Given a list of package purls, return a mapping of {expression: count
+    of occurences}
+    """
+    return Counter(package_purls)
```

Comment:
This is missing a training line return

Reply:
You mean there should be a blank line before the `return` statement?

Reply:
yes, the convention is that all Python sources (and in general most source code text) should end up with a line return. See https://www.reddit.com/r/Python/comments/1zjugg/question_why_does_pep8_recommend_leaving_a_blank/ for some commentary... that a minor nitpicking though! ;)

Reply:
Thanks for pointing out. I've made the changes!